google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineRunConfigurationEditor.java

```diff
@@ -114,11 +114,6 @@ protected void resetEditorFrom(CommonModel commonModel) {
     final AppEngineServerModel serverModel = (AppEngineServerModel) commonModel.getServerModel();
     final Artifact artifact = serverModel.getArtifact();
     myArtifactComboBox.setSelectedItem(artifact);
-    if (artifact == null && myArtifactComboBox.getItemCount() == 1) {
```

Comment:
There is no NPE when you run the config that I know of.

getItemCount() here returns 1 both for Java and Java EE projects, because we're in the UI editor and, therefore, at least one run config must exist.

What returns 1 for Java EE projects is the list size in the following line:

https://github.com/GoogleCloudPlatform/google-cloud-intellij/blob/master/google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/facet/impl/AppEngineUltimateWebIntegration.java#L113

The list isn't empty, so we move on with the template configuration that doesn't have an artifact associated. (If the list were empty, we would associate an artifact right after)

Then, you would open the editor which, by default, automatically selects the artifact if the project only contains one artifact, which makes the error go away. However, when the UI is closed, the artifact was actually never persisted to the config, so the error comes back.

My fix is to not automatically pre-select the artifact, but let the user do it, so they will have done something for it to go away. We could also just persist the artifact when we automatically pre-select it, but then the error would have gone away without the users having done nothing out of the ordinary, which is a slightly worse experience imo..